### PR TITLE
Adding maximize to rprop

### PR DESCRIPTION
**Pull request by zaxtax**
Added the maximize flag #68052 to rprop optimizer and updates the respective tests.

### CODE REVIEW

File: test/test_optim.py [hunk 1]
```diff
@@ -880,15 +880,18 @@ def test_asgd(self):
             with self.assertRaisesRegex(ValueError, "Invalid weight_decay value: -0.5"):
                 optimizer(None, lr=1e-2, weight_decay=-0.5)
 
+    @skipIfRocm
```
Comment (albanD) [inline test/test_optim.py:883]: There are numerical precision issues on rocm I suppose?

**Review by albanD — APPROVED** [review]
Sounds good!

File: test/test_optim.py (hunk 1 above)
Comment (zaxtax) [reply test/test_optim.py:883]: Yep.

**Comment by rgommers** [comment]
The one test failure on `win-vs2019-cuda11.6-py3 / test (default, 2, 5, windows.8xlarge.nvidia.gpu)` is relevant:

```
======================================================================
FAIL [2.729s]: test_rprop (__main__.TestOptim)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\actions-runner\_work\pytorch\pytorch\build\win_tmp\build\torch\testing\_internal\common_utils.py", line 974, in wrapper
    fn(*args, **kwargs)
  File "test_optim.py", line 901, in test_rprop
    self._test_basic_cases(
  File "test_optim.py", line 262, in _test_basic_cases
    self._test_state_dict(
  File "test_optim.py", line 243, in _test_state_dict
    self.assertEqual(weight, weight_cuda)
  File "C:\actions-runner\_work\pytorch\pytorch\build\win_tmp\build\torch\testing\_internal\common_utils.py", line 2366, in assertEqual
    assert_equal(
  File "C:\actions-runner\_work\pytorch\pytorch\build\win_tmp\build\torch\testing\_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 5 / 50 (10.0%)
Greatest absolute difference: 2.3290514945983887e-05 at index (5, 0) (up to 1e-05 allowed)
Greatest relative difference: 0.00011316381109651172 at index (5, 0) (up to 1.3e-06 allowed)
```
Probably just needs a small bump in max allowed `atol`?

**Pull request closed without merging**